src/sqlitedb.cpp

```diff
@@ -279,30 +286,94 @@ bool DBBrowserDB::tryEncryptionSettings(const QString& filename, bool* encrypted
         {
             sqlite3_finalize(vm);
 #ifdef ENABLE_SQLCIPHER
-            delete cipherSettings;
-            cipherSettings = new CipherDialog(nullptr, false);
-            if(cipherSettings->exec())
+            // Being in a while loop, we don't want to check the same file multiple times
+            if (!isDotenvChecked) {
+                QFile databaseFile(filePath);
+                QFileInfo databaseFileInfo(databaseFile);
+
+                QString databaseDirectoryPath = databaseFileInfo.dir().path();
+                QString databaseFileName(databaseFileInfo.fileName());
+
+                QString dotenvFilePath = databaseDirectoryPath + "/.env";
+                QSettings::Format dotenvFormat = Settings::getDotenvFormat();
+                QSettings dotenv(dotenvFilePath, dotenvFormat);
+
+                QVariant passwordValue = dotenv.value(databaseFileName);
+
+                bool foundPassword = !passwordValue.isNull();
+
+                isDotenvChecked = true;
+
+                if (foundPassword)
+                {
+                    QString password = passwordValue.toString();
+
+                    QVariant keyFormatValue = dotenv.value(databaseFileName + "_keyFormat", QVariant(CipherSettings::KeyFormats::Passphrase));
+                    CipherSettings::KeyFormats keyFormat = CipherSettings::getKeyFormat(keyFormatValue.toInt());
+
+                    QVariant pageSizeValue = dotenv.value(databaseFileName + "_pageSize", QVariant(CipherSettings::defaultPageSize));
+                    int pageSize = pageSizeValue.toInt();
+
+                    cipherSettings = new CipherSettings();
+
+                    cipherSettings->setKeyFormat(keyFormat);
+                    cipherSettings->setPassword(password);
+                    cipherSettings->setPageSize(pageSize);
+
+                    // Close and reopen database first to be in a clean state after the failed read attempt from above
```

Comment:
Can you change the code here (maybe by introducing some flag or moving it into a separate function) to avoid duplicating the lines below? Everything from this line until the ```*encrypted = true;``` line is basically a copy of the code below and I feel like this code is long and complex enough to not be twice in the code base - especially because it might be changed in the future.